demo/src/App.tsx

```diff
@@ -23,20 +23,33 @@ let roomId: string;
 
 interface AppProps {}
 
-const App: FC<AppProps> = () => (
-  <ChatRoomProvider
-    id={roomId}
-    release={true}
-    attach={true}
-    options={RoomOptionsDefaults}
-  >
-    <div style={{ display: 'flex', justifyContent: 'space-between', width: '800px', margin: 'auto' }}>
-      <Chat />
-      <div style={{ display: 'flex', flexDirection: 'column' }}>
-        <UserPresenceComponent />
-        <OccupancyComponent />
+const App: FC<AppProps> = () => {
+  const [roomIdState, setRoomId] = useState(roomId);
+  const updateRoomId = (newRoomId: string) => {
+    const params = new URLSearchParams(window.location.search);
+    params.set('room', newRoomId);
+    history.replaceState(null, '', '?' + params.toString());
```

Comment:
push so back button takes you to previous room?

Reply:
Good shout, done that